hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/WriteOperationHelper.java

```diff
@@ -236,7 +236,8 @@ public String initiateMultiPartUpload(String destKey) throws IOException {
    * @return the result of the operation.
    * @throws IOException on problems.
    */
-  @Retries.RetryTranslated
+  @Retries.RetryTranslated("For complete MPU; post-complete actions are "
```

Comment:
I think I have this wrong.  Now my impression is I shouldn't have changed this annotation.  This entire operation is effectively RetryTranslated.  This method itself retries and translates exceptions on complete-MPU and the finishedWrite() method retries and translates exceptions internal to itself (via the retrying and translating occurring in DynamoDBMetadataStore).  Is that how you would think of it?

Reply:
if AWS exceptions are being caught & stuck through translateException, they are retry-translated. The key thing is: by the time any operation gets to the public FS APIs, they must be translated. Double translate is harmless. It is retry-on-retry

Reply:
Thanks.  Looks like I can just revert the change on this line here and just leave it as it was before since it already said Retries.RetryTranslated and the extra text I added isn't helping any.

Reply:
Updated in https://github.com/apache/hadoop/pull/666/commits/8ff542741ab3386a9ce967fb2f36fab2265413f4